File: ruby/compete/code_golf/power-of-thor.rb
a,b,c,d=gets.split.map &:to_i
loop{puts (b<d ?(d-=1;'N'):d<b ?(d+=1;'S'):'')+(a<c ?(c-=1;'W'):c<a ?(c+=1;'E'):'')}

File: perl/compete/code_golf/power-of-thor.pl
$|=1;my($a,$b,$c,$d)=split(/ /,<STDIN>);while(1){my$e;$b<$d?($d--,$e='N'):$d<$b?($d++,$e='S'):0;$a<$c?($c--,$e.='W'):$c<$a?($c++,$e.='E'):0;print "$e\n"}
